components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -6322,4 +6328,210 @@ public static boolean isExistingUser(String userName, String tenantDomain) throw
                 realmService.getTenantUserRealm(tenantId).getUserStoreManager();
         return userStoreManager.isExistingUser(MultitenantUtils.getTenantAwareUsername(userName));
     }
+
+    /**
+     * Get token binding value from the request headers.
+     *
+     * @param oAuth2AccessTokenReqDTO OAuth2AccessTokenReqDTO
+     * @param cookieName              Cookie name
+     * @return Token binding value
+     */
+    public static Optional<String> getTokenBindingValue(OAuth2AccessTokenReqDTO oAuth2AccessTokenReqDTO,
```

Comment:
Does this need to be a public util method or should go under session binder? Are we using this method in different places?